### DISCUSSION TITLE
Lock while editing Mysql procedures

### DISCUSSION THREAD

**Discussion by davideio87** [dbeaver/dbeaver#15137] [Q&A] [answered]
Hello,
I have a "problem" on my 21.1 version of dbeaver, but I'm facing it since a lot of previous versions.

I use it to work on a "Development" Mysql database, so the Auto commit is disabled

**Here is the issue:** when I open a stored procedure (by double click) and edit it directly inside the Properties editor, most of the time ( let's say 90%) I can't save my edit because when I press Save-> Persist the session stay pending, with "Waiting for stored procedure metadata lock" status. In the other 10% it instantly work

The only way to move on is to open the session manager via PHPMyAdmin, find the session that is locking the procedure and killing it.
The problem is that the locking session is.. myself! It is the session that dbeaver creates when browsing the list of procedure
I'm 100% sure about this because I'm the only one working on this database and no other tabs or process are actually using that procedure.

Am I missing some obvious setting? since is a dev environment I want to keep autocommit disabled

Thanks

**Comment by Matvey16** [comment, marked as answer]
@davideio87 Have you tried unchecking an option to open separate connection for metadata read?

![image](https://user-images.githubusercontent.com/82543000/147662224-3d63189a-9677-42ab-a8af-9e95794a870c.png)

**Comment by davideio87** [comment]
It seems exactly what I was looking for!

Thanks a lot!

**Comment by guptaamol90** [comment]
I am trying to persist changes for MySQL  8 Stored Procedure  in dbeaver Version 23.3.5.202402181738, the call for stored procedure is stuck forever. I have to manually kill the process and I did manually drop and recreate the procedure after which the call on sp works fine.
I have tried the setting mentioned above and I have unchecked it. But it is not helping. Can you please check this issue urgently???
